### PR TITLE
Feature/cqlhs 49 User Auth

### PR DESCRIPTION
**Pull request by charkour**
This provides functionality for the user to sign in and be authenticated by the backend.

### CODE REVIEW

**Comment by charkour** [comment]
The tests are still a work in progress, but there will be more in the future!

**Comment by jonathanchaffer** [comment]
Can we store the user in cookies so the user stays logged in on refresh? [react-cookies](https://www.npmjs.com/package/react-cookies) should be helpful for this.

File: ClientApp/src/components/pages/LoginPage/LoginPage.tsx [hunk 1]
```diff
@@ -84,7 +73,6 @@ export const LoginPage: React.FC = () => {
           <Form.Group controlId="username">
             <Form.Label>Username</Form.Label>
             <Form.Control
-              disabled={isLoading}
```
Comment (jonathanchaffer) [inline ClientApp/src/components/pages/LoginPage/LoginPage.tsx:87]: why is this removed?

Comment (charkour) [reply ClientApp/src/components/pages/LoginPage/LoginPage.tsx:87]: Oops, missed that.

**Comment by charkour** [comment]
Yeah, I can add that in another PR after the demo.
